### PR TITLE
Added bash script to intall Kafka and Litmus

### PR DESCRIPTION
**Pull request by tonyxrmdavidson**
https://issues.redhat.com/browse/MGDSTRM-954

This is the 1st part of the task to create a litmus testing pipeline that will make it more accessible to the team members without needing to set up and run locally.

This PR merges a bash script that will install Kafka and Litmus onto an existing OSD cluster.

### CODE REVIEW

**Comment by tonyxrmdavidson** [comment]
The last line of the commit is deliberately commented out. I just want to get this in so I can test the pipeline that calls the script.

**Comment by tonyxrmdavidson** [comment]
Output from locally run script.   `Tony@WorkMac: ~/kafka-monitoring-stuff (chaosSchedulerUninstallFix) $ ./install/resources/testing-chaos/scripts/run-litmus-test.sh
Logging into openshift cluster
Logged into "https://api.tdavidso.pdxt.s1.devshift.org:6443" as "kube:admin" using the token provided.

You have access to 83 projects, the list has been suppressed. You can list all projects with 'oc projects'

Using project "managed-services-monitoring-grafana".
make -C resources/operator all
create namespace: kafka-operator
namespace/kafka-operator created
create operatorgroup: kafka-operator
operatorgroup.operators.coreos.com/operatorgroup created
create subscription: kafka-operator
subscription.operators.coreos.com/my-strimzi-kafka-operator created
Waiting for Kafka CRD to be created
NAME                      CREATED AT
kafkas.kafka.strimzi.io   2021-01-05T13:14:47Z
Done installing strimzi-kafka-operator
make -C resources/kafka all
create namespace: kafka-cluster
namespace/kafka-cluster created
create kafka cluster: kafka-cluster
kafka.kafka.strimzi.io/my-cluster created
configmap/kafka-logging unchanged
Done installing kafka cr
make -C resources/monitoring-global all
create namespace: managed-services-monitoring-global
namespace/managed-services-monitoring-global unchanged
create thanos receiver: kafka-thanos-receiver
deployment.apps/kafka-thanos-receiver unchanged
service/kafka-thanos-receiver-service unchanged
route.route.openshift.io/kafka-thanos-receiver-route unchanged
create thanos querier: kafka-thanos-querier
deployment.apps/kafka-thanos-querier unchanged
service/kafka-thanos-querier-service unchanged
route.route.openshift.io/kafka-thanos-querier-route unchanged
create grafana: kafka-grafana-global
subscription.operators.coreos.com/kafka-grafana-global unchanged
operatorgroup.operators.coreos.com/kafka-grafana-global unchanged
Waiting for Grafana CRD to be created
NAME                       CREATED AT
grafanas.integreatly.org   2021-01-05T13:15:25Z
secret/grafana-k8s-proxy configured
clusterrole.rbac.authorization.k8s.io/grafana-oauth-proxy-cluster-role unchanged
clusterrolebinding.rbac.authorization.k8s.io/global-grafana-oauth-proxy-cluster-role-binding unchanged
grafana.integreatly.org/kafka-grafana-global unchanged
grafanadatasource.integreatly.org/kafka-grafana-global unchanged
creating global kafka dashboards
grafanadashboard.integreatly.org/fleet unchanged
grafanadashboard.integreatly.org/logs unchanged
grafanadashboard.integreatly.org/kafka-namespace unchanged
grafanadashboard.integreatly.org/kafka-pod unchanged
grafanadashboard.integreatly.org/kafka-pv unchanged
create loki: kafka-loki-global
deployment.apps/kafka-loki-global unchanged
configmap/kafka-loki-global-config unchanged
service/kafka-loki-global unchanged
route.route.openshift.io/kafka-loki-global unchanged
grafanadatasource.integreatly.org/kafka-loki-global unchanged
Done installing fleet wide monitoring stack
# create namepsace
Error from server (AlreadyExists): project.project.openshift.io "managed-services-monitoring-grafana" already exists
make: [install/monitoring/cluster] Error 1 (ignored)
# create operatorgroup
operatorgroup.operators.coreos.com/grafana-operator unchanged
# create subscription
subscription.operators.coreos.com/grafana-operator unchanged
Waiting for Grafana CRD to be created
NAME                       CREATED AT
grafanas.integreatly.org   2021-01-05T13:15:25Z
# create grafana CR
secret/grafana-k8s-proxy configured
clusterrole.rbac.authorization.k8s.io/grafana-oauth-proxy-cluster-role unchanged
clusterrolebinding.rbac.authorization.k8s.io/cluster-grafana-oauth-proxy-cluster-role-binding unchanged
grafana.integreatly.org/grafana unchanged
Waiting for grafana pod to be ready
pod/grafana-deployment-644c479d74-pp6tn
create namespace: managed-services-monitoring-prometheus
namespace/managed-services-monitoring-prometheus unchanged
installing prometheus: kafka-prometheus
subscription.operators.coreos.com/kafka-prometheus unchanged
operatorgroup.operators.coreos.com/kafka-prometheus unchanged
serviceaccount/kafka-prometheus unchanged
clusterrole.rbac.authorization.k8s.io/kafka-prometheus unchanged
clusterrolebinding.rbac.authorization.k8s.io/kafka-prometheus unchanged
prometheus.monitoring.coreos.com/kafka-prometheus unchanged
podmonitor.monitoring.coreos.com/cluster-operator-metrics unchanged
podmonitor.monitoring.coreos.com/kafka-metrics unchanged
route.route.openshift.io/kafka-prometheus unchanged
Waiting for the prometheus operator to be ready
pod/prometheus-operator-789d89d666-nlt96
enabling federation of openshift metrics
secret/additional-scrape-configs configured
creating grafana data souroce
grafanadatasource.integreatly.org/grafanadatasource unchanged
grafanadatasource.integreatly.org/observatorium-metrics unchanged
grafanadatasource.integreatly.org/observatorium-logs unchanged
creating prometheusrules
prometheusrule.monitoring.coreos.com/kafka-prometheus-rules unchanged
finished setup of cluster metrics collection
creating kafka cluster dashboards
grafanadashboard.integreatly.org/strimzi-kafka unchanged
grafanadashboard.integreatly.org/strimzi-kafka-exporter unchanged
grafanadashboard.integreatly.org/strimzi-operators unchanged
grafanadashboard.integreatly.org/strimzi-zookeeper unchanged
grafanadashboard.integreatly.org/slis unchanged
create promtail: kafka-promtail
serviceaccount/kafka-promtail unchanged
clusterrole.rbac.authorization.k8s.io/kafka-promtail unchanged
clusterrolebinding.rbac.authorization.k8s.io/kafka-promtail unchanged
configmap/kafka-promtail unchanged
daemonset.apps/kafka-promtail configured
finished setup of cluster promtail
done installing kafka monitoring demo
checking kafka-operator status
KAFKA Operator pod is Running
checking kafka-cluster pods status
KAFKA cluster pods are Running
./install/resources/testing-chaos/scripts/run-litmus-test.sh: line 28: one: command not found
instlling litmus operator
create operator: litmus
namespace/litmus created
deployment.apps/chaos-operator-ce created
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
customresourcedefinition.apiextensions.k8s.io/chaosengines.litmuschaos.io configured
customresourcedefinition.apiextensions.k8s.io/chaosexperiments.litmuschaos.io configured
customresourcedefinition.apiextensions.k8s.io/chaosresults.litmuschaos.io unchanged
serviceaccount/litmus-admin created
clusterrole.rbac.authorization.k8s.io/litmus-admin unchanged
clusterrolebinding.rbac.authorization.k8s.io/litmus-admin unchanged
checking litmus-operator pod status
Litmus operator pod is Running
installing the chaosscheduler operator
create operator: chaosscheduler
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
customresourcedefinition.apiextensions.k8s.io/chaosschedules.litmuschaos.io configured
deployment.apps/chaos-scheduler created
serviceaccount/scheduler created
Warning: rbac.authorization.k8s.io/v1beta1 ClusterRole is deprecated in v1.17+, unavailable in v1.22+; use rbac.authorization.k8s.io/v1 ClusterRole
clusterrole.rbac.authorization.k8s.io/scheduler unchanged
Warning: rbac.authorization.k8s.io/v1beta1 ClusterRoleBinding is deprecated in v1.17+, unavailable in v1.22+; use rbac.authorization.k8s.io/v1 ClusterRoleBinding
clusterrolebinding.rbac.authorization.k8s.io/scheduler unchanged
checking chaossscheduler pod status
Litmus chaosscheduler pod is Running`

File: install/resources/testing-chaos/scripts/run-litmus-test.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      echo "KAFKA cluster pods are Running"
+      break
+    fi
+    sleep 5
+done
+
+cd resources/testing-chaos/
+
+echo "instlling litmus operator"
+make create/operator/litmus
```
Comment (Rajagopalan-Ranganathan) [inline install/resources/testing-chaos/scripts/run-litmus-test.sh:32]: Can check for the return value of "make" and log error if make fails.

File: install/resources/testing-chaos/scripts/run-litmus-test.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+echo "checking litmus-operator pod status"
+
+for (( i=0; i<120; i++ )) do
+    LITMUS_POD_RUNNING=`oc get pods -n litmus | grep -c Running` 
+    if [[ $LITMUS_POD_RUNNING = 1 ]]; then
```
Comment (Rajagopalan-Ranganathan) [inline install/resources/testing-chaos/scripts/run-litmus-test.sh:38]: Same as above comment with respect to running count.

File: install/resources/testing-chaos/scripts/run-litmus-test.sh [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    fi
+    sleep 5
+done
+
+echo "installing the chaosscheduler operator"
+make create/operator/chaosscheduler
+
+echo "checking chaossscheduler pod status"
+for (( i=0; i<120; i++ )) do
+    LITMUS_POD_RUNNING=`oc get pods -n litmus | grep -c Running`
```
Comment (Rajagopalan-Ranganathan) [inline install/resources/testing-chaos/scripts/run-litmus-test.sh:50]: Same as above comment with respect to running count.

File: install/resources/testing-chaos/scripts/run-litmus-test.sh [hunk 4]
```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+echo "Logging into openshift cluster"
+oc login --server=$OSD_CLUSTER_API --token=$OSD_API_TOKEN --insecure-skip-tls-verify
+cd install
+make all
```
Comment (Rajagopalan-Ranganathan) [inline install/resources/testing-chaos/scripts/run-litmus-test.sh:6]: Check the return code of make command. Log error and exit/continue/retry as necessary

File: install/resources/testing-chaos/scripts/run-litmus-test.sh [hunk 5]
```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+echo "Logging into openshift cluster"
+oc login --server=$OSD_CLUSTER_API --token=$OSD_API_TOKEN --insecure-skip-tls-verify
```
Comment (Rajagopalan-Ranganathan) [inline install/resources/testing-chaos/scripts/run-litmus-test.sh:4]: Check the return of oc login command, if the login fails - we must log error and exit.

File: install/resources/testing-chaos/scripts/run-litmus-test.sh [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    LITMUS_POD_RUNNING=`oc get pods -n litmus | grep -c Running` 
+    if [[ $LITMUS_POD_RUNNING = 1 ]]; then
+      echo "Litmus operator pod is Running"
+      break
+    fi
+    sleep 5
+done
+
+echo "installing the chaosscheduler operator"
+make create/operator/chaosscheduler
```
Comment (Rajagopalan-Ranganathan) [inline install/resources/testing-chaos/scripts/run-litmus-test.sh:46]: Check the return value of make - log error/exit/retry as necessary

File: install/resources/testing-chaos/scripts/run-litmus-test.sh [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+echo "checking kafka-operator pod status"
+for (( i=0; i<120; i++ )) do
+    KAFKA_POD_RUNNING=`oc get pods -n kafka-operator | grep -c Running` 
+    if [[ $KAFKA_POD_RUNNING = 1 ]]; then
+      echo "KAFKA Operator pod is Running"
+      break
+    fi
+    sleep 5
+done
+
```
[the remark this answers is not in the document]
Comment (tonyxrmdavidson) [reply install/resources/testing-chaos/scripts/run-litmus-test.sh:17]: I agree,  will do that on the next iteration

File: install/resources/testing-chaos/scripts/run-litmus-test.sh [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+echo "checking kafka-cluster pods status"
+
+for (( i=0; i<180; i++ )) do
+    KAFKA_CLUSTER_PODS_RUNNING=`oc get pods -n kafka-cluster | grep -c Running` 
+    if [[ $KAFKA_CLUSTER_PODS_RUNNING = 7 ]]; then
```
[the remark this answers is not in the document]
Comment (tonyxrmdavidson) [reply install/resources/testing-chaos/scripts/run-litmus-test.sh:22]: I think you correct here.  I think I will get all all available pods and check they are all in a running state.

File: install/resources/testing-chaos/scripts/run-litmus-test.sh [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+      echo "KAFKA cluster pods are Running"
+      break
+    fi
+    sleep 5
+done
+
```
[the remark this answers is not in the document]
Comment (tonyxrmdavidson) [reply install/resources/testing-chaos/scripts/run-litmus-test.sh:28]: 👍🏽

**Pull request merged**